r/geoarrow/R/arrow-compat.R

```diff
@@ -0,0 +1,177 @@
+
+as_arrow_array.geoarrow_vctr <- function(x, ..., type = NULL) {
+  chunked <- as_chunked_array.geoarrow_vctr(x, ..., type = type)
+  if (chunked$num_chunks == 1) {
+    chunked$chunk(0)
+  } else {
+    arrow::as_arrow_array(chunked)
+  }
+}
+
+as_chunked_array.geoarrow_vctr <- function(x, ..., type = NULL) {
+  if (is.null(type)) {
+    type <- arrow::as_data_type(attr(x, "schema", exact = TRUE))
+    chunks <- attr(x, "chunks", exact = TRUE)
+  } else {
+    stream <- as_nanoarrow_array_stream(x, schema = as_nanoarrow_schema(type))
+    chunks <- nanoarrow::collect_array_stream(stream, validate = FALSE)
+    type <- arrow::as_data_type(type)
+  }
+
+  schema <- as_nanoarrow_schema(type)
+  arrays <- vector("list", length(chunks))
+  for (i in seq_along(arrays)) {
+    tmp_schema <- nanoarrow::nanoarrow_allocate_schema()
+    nanoarrow::nanoarrow_pointer_export(schema, tmp_schema)
+    tmp_array <- nanoarrow::nanoarrow_allocate_array()
+    nanoarrow::nanoarrow_pointer_export(chunks[[i]], tmp_array)
+    arrays[[i]] <- arrow::Array$import_from_c(tmp_array, tmp_schema)
+  }
+
+  arrow::ChunkedArray$create(!!!arrays, type = type)
```

Comment:
I assume the following isn't equivalent? My lack of knowledge of arrow and nanoarrow internals is probably obvious with this question.
```r
arrow::chunked_array(!!!chunks, type = type)
```

Reply:
I don't understand what these lines are for? Can we just create the array from `chunks`?

https://github.com/geoarrow/geoarrow-c/blob/a9293f36ef9ca047baadcf86679f2304542fcd52/r/geoarrow/R/arrow-compat.R#L26-L33